src/Symfony/Component/DependencyInjection/Attribute/Autoconfigure.php

```diff
@@ -20,16 +20,16 @@
 class Autoconfigure
 {
     /**
-     * @param array<array-key, array<array-key, mixed>>|string[]|null $tags         The tags to add to the service
-     * @param array<array<mixed>>|null                                $calls        The calls to be made when instantiating the service
-     * @param array<string, mixed>|null                               $bind         The bindings to declare for the service
-     * @param bool|string|null                                        $lazy         Whether the service is lazy-loaded
-     * @param bool|null                                               $public       Whether to declare the service as public
-     * @param bool|null                                               $shared       Whether to declare the service as shared
-     * @param bool|null                                               $autowire     Whether to declare the service as autowired
-     * @param array<string, mixed>|null                               $properties   The properties to define when creating the service
-     * @param array<class-string, string>|string|null                 $configurator A PHP function, reference or an array containing a class/Reference and a method to call after the service is fully initialized
-     * @param string|null                                             $constructor  The public static method to use to instantiate the service
+     * @param array<array-key, array<array-key, mixed>>|string[]|null                      $tags         The tags to add to the service
+     * @param array<array<mixed>>|null                                                     $calls        The calls to be made when instantiating the service
+     * @param array<string, mixed>|null                                                    $bind         The bindings to declare for the service
+     * @param bool|string|null                                                             $lazy         Whether the service is lazy-loaded
+     * @param bool|null                                                                    $public       Whether to declare the service as public
+     * @param bool|null                                                                    $shared       Whether to declare the service as shared
+     * @param bool|null                                                                    $autowire     Whether to declare the service as autowired
+     * @param array<string, mixed>|null                                                    $properties   The properties to define when creating the service
+     * @param array{non-empty-string|class-string, non-empty-string}|non-empty-string|null $configurator A PHP function, reference or an array containing a class/Reference and a method to call after the service is fully initialized
```

Comment:
let's remove the `empty-` prefixes, they don't provide much
and let's merge class-string into string:
```suggestion
     * @param array{string, string}|string|null $configurator A PHP function, reference or an array containing a class/Reference and a method to call after the service is fully initialized
```

Reply:
Hello @nicolas-grekas, thanks for the review!

I thought a non-empty string (as the first argument) would make sense since an empty string would trigger a RuntimeException in the PhpDumper :) . But anyway, using an empty string here doesn’t really make sense, so…

I've applied the changes you suggested. :)